astropy/cosmology/funcs/comparison.py

```diff
@@ -0,0 +1,439 @@
+    -------
+    callable[..., Any]
+        Wrapped `pyfunc`, as described above.
+
+    Notes
+    -----
+    All decorated functions should add the following to 'Parameters'.
+
+    format : bool or None or str or array-like thereof, optional keyword-only
+        Whether to allow the arguments to be converted to a |Cosmology|. This
+        allows, e.g. a |Table| to be given instead a Cosmology. `False`
+        (default) will not allow conversion. `True` or `None` will, and will use
+        the auto-identification to try to infer the correct format. A `str` is
+        assumed to be the correct format to use when converting. Note 'format'
+        is broadcast as an object array to match the shape of 'cosmos' so
+        'format' cannot determine the output shape.
+    """
+    sig = inspect.signature(pyfunc)
+    nin = sum([p.kind == 0 for p in sig.parameters.values()])
+
+    # Make wrapper function that parses cosmology-like inputs
+    @functools.wraps(pyfunc)
+    def wrapper(*cosmos: Any, format: _FormatsT = False, **kwargs: Any) -> bool:
+        if len(cosmos) > nin:
+            raise TypeError
+        # Parse cosmologies to format. Only do specified number.
+        cosmos = _parse_formats(*cosmos, format=format)
+        # Evaluate pyfunc, erroring if didn't match specified number.
+        result = wrapper.__wrapped__(*cosmos, **kwargs)
+        # Return, casting to correct type casting is possible.
+        return result
+
+    return wrapper
+
+
+##############################################################################
+# COMPARISON FUNCTIONS
+
+
+@_comparison_decorator
+def cosmology_equal(cosmo1: Any, cosmo2: Any, /, *, allow_equivalent: bool=False) -> bool:
+    r"""Return element-wise equality check on the cosmologies.
+
+    .. note::
+
+        Cosmologies are currently scalar in their parameters.
+
+    Parameters
+    ----------
+    cosmo1, cosmo2 : |Cosmology|-like
+        The objects to compare. Must be convertible to |Cosmology|, as specified
+        by ``format``.
+
+    format : bool or None or str or tuple thereof, optional keyword-only
+        Whether to allow the arguments to be converted to a |Cosmology|. This
+        allows, e.g. a |Table| to be given instead a Cosmology. `False`
+        (default) will not allow conversion. `True` or `None` will, and will use
+        the auto-identification to try to infer the correct format. A `str` is
+        assumed to be the correct format to use when converting. Note 'format'
+        is broadcast as an object array to match the shape of 'cosmos' so
+        'format' cannot determine the output shape.
+
+    allow_equivalent : bool, optional keyword-only
+        Whether to allow cosmologies to be equal even if not of the same class.
+        For example, an instance of :class:`~astropy.cosmology.LambdaCDM` might
+        have :math:`\Omega_0=1` and :math:`\Omega_k=0` and therefore be flat,
+        like :class:`~astropy.cosmology.FlatLambdaCDM`.
```

Comment:
I'm only highlighting the suggested change here, but there are more instances.
```suggestion
        For example, an instance of |LambdaCDM| might
        have :math:`\Omega_0=1` and :math:`\Omega_k=0` and therefore be flat,
        like |FlatLambdaCDM|.
```

Reply:
👍 ✅